[CI] HeapAttackIT testSortByManyLongsTooMuchMemoryAsync failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3484 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/acuczfm56qo2k)
- [elasticsearch-periodic-platform-support #3484 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/73qqehhdgu4jc)
- [elasticsearch-periodic #3506 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/kxtxik5tropmq)
- [elasticsearch-periodic #3506 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/jalldq4fxyffk)
- [elasticsearch-periodic-platform-support #3480 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/lldw6zvmacmtw)
- [elasticsearch-periodic-platform-support #3480 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/35vxxhgijzvoo)
- [elasticsearch-periodic-platform-support #3480 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/eht6liv6kt6ck)
- [elasticsearch-periodic-platform-support #3480 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/b7ewjekakop6k)
- [elasticsearch-periodic-platform-support #3480 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/bdwcptiqvvbd4)
- [elasticsearch-periodic #3502 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/xzns2ohlkr2em)

**Reproduction Line:**
```
./gradlew ":test:external-modules:test-esql-heap-attack:javaRestTest" --tests "org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT.testSortByManyLongsTooMuchMemoryAsync" -Dtests.seed=87D5A91AE1F0234A -Dtests.locale=sq-AL -Dtests.timezone=Africa/Timbuktu -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testSortByManyLongsTooMuchMemoryAsync'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
    status: <429>
     error: a map containing
bytes_wanted: <322135864>
      reason: "[request] Data too large, data for [topn] would be [322135864/307.2mb], which is larger than the limit of [322122547/307.1mb]"
  durability: "TRANSIENT"
        type: "circuit_breaking_exception"
 bytes_limit: <322122547>
  root_cause: a list containing
             0: <{bytes_wanted=322135864, reason=[request] Data too large, data for [topn] would be [322135864/307.2mb], which is larger than the limit of [322122547/307.1mb], durability=TRANSIENT, suppressed=[{reason=1 further exceptions were dropped, type=exception}, {reason=cancelled on failure, type=task_cancelled_exception}, {reason=cancelled on failure, type=task_cancelled_exception}], type=circuit_breaking_exception, bytes_limit=322122547}>
  suppressed: <unexpected> but was <[{reason=1 further exceptions were dropped, type=exception}, {reason=cancelled on failure, type=task_cancelled_exception}, {reason=cancelled on failure, type=task_cancelled_exception}]>
```

**Issue Reasons:**
- [main] 3 consecutive failures in step rhel-7_platform-support-unix
- [main] 3 consecutive failures in step concurrent-search-tests
- [main] 25 failures in test testSortByManyLongsTooMuchMemoryAsync (9.7% fail rate in 259 executions)
- [main] 4 failures in step rhel-7_platform-support-unix (57.1% fail rate in 7 executions)
- [main] 4 failures in step concurrent-search-tests (57.1% fail rate in 7 executions)
- [main] 3 failures in step amazonlinux-2023_platform-support-aws (42.9% fail rate in 7 executions)
- [main] 4 failures in step rocky-8_platform-support-unix (57.1% fail rate in 7 executions)
- [main] 2 failures in step oraclelinux-7_platform-support-unix (28.6% fail rate in 7 executions)
- [main] 8 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 8 executions)
- [main] 4 failures in pipeline elasticsearch-periodic (50.0% fail rate in 8 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
+++ b/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
@@ -162,7 +162,8 @@ public class HeapAttackIT extends ESRestTestCase {
             matchesMap().entry("status", 429)
                 .entry(
                     "error",
-                    matchesMap().entry("bytes_wanted", greaterThan(1000))
+                    matchesMap().extraOk()
+                        .entry("bytes_wanted", greaterThan(1000))
                         .entry("reason", matchesRegex("\\[request] Data too large, data for \\[topn] would .+"))
                         .entry("durability", "TRANSIENT")
                         .entry("type", "circuit_breaking_exception")

